src/pages/api/answer.ts

```diff
@@ -1,23 +1,41 @@
 import {NextApiRequest, NextApiResponse} from 'next'
 import {convertkitAxios} from 'utils/axios-convertkit-api'
 import fetchConvertkitSubscriberFromServerCookie from 'utils/fetch-convertkit-subscriber'
+import isEmpty from 'lodash/isEmpty'
 
 const answer = async (req: NextApiRequest, res: NextApiResponse) => {
   if (req.method === 'POST') {
     try {
       console.log(req.body)
-      const {tagId} = req.body
+      const {tagId, quiz} = req.body
       const cookieHeader = req.headers.cookie as string
-      const [
-        subscriber,
-        ckCookie,
-      ] = await fetchConvertkitSubscriberFromServerCookie(cookieHeader)
+      const [subscriber, ckCookie] =
+        await fetchConvertkitSubscriberFromServerCookie(cookieHeader)
 
+      // Subscribe user to tagId
       await convertkitAxios.post(`/tags/${tagId}/subscribe`, {
         api_key: process.env.NEXT_PUBLIC_CONVERTKIT_TOKEN,
         email: subscriber.email_address,
       })
 
+      // Create Quiz answers field if it doesn't exist yet
+      if (!subscriber.fields.quiz_answers) {
+        await convertkitAxios.post(`/custom_fields`, {
+          api_secret: process.env.CONVERTKIT_API_SECRET,
+          label: 'Quiz answers',
+        })
+      }
+
+      // Update Quiz answers field with submitted answer along with a question id
+      await convertkitAxios.put(`/subscribers/${subscriber.id}`, {
+        api_secret: process.env.CONVERTKIT_API_SECRET,
+        fields: {
+          quiz_answers: !isEmpty(subscriber.fields.quiz_answers)
+            ? `${JSON.stringify(quiz)},${subscriber.fields.quiz_answers}`
+            : JSON.stringify(quiz),
+        },
```

Comment:
custom fields are plain text and this results in following:
`{"id":"multipleCorrect","answer":["two","four"]},{"id":"trueFalse","answer":"false"},{"id":"essay","answer":"test"},{"id":"essay","answer":"test"}`

IDs don't make sense in this example but in reality they are more descriptive.

Reply:
for fields, I'd consider using a pattern like `JJS-quiz-question-name` or `ER-survey-question-name` where the value is the answer. If you wanted to codify the type it could be `JJS-mc-question-name` where the type is the second aspect (`mc` == multiple choice)

Reply:
store the short name (JJS) as env var or make it part of questions object?